File: Model/src/main/php/org/xdi/oxauth/model/common/Prompt.php
<?php
namespace org\xdi\oxauth\model\common;

/**
 * An ASCII string values that specifies whether the Authorization Server
 * prompts the End-User for re-authentication and consent.
 *
 * @author Gabin Dongmo Date: 29/03/2013
 */
class Prompt implements HasParamName {
    
    /**
     * The Authorization Server MUST NOT display any authentication or
     * consent user interface pages. An error is returned if the End-User
     * is not already authenticated or the Client does not have pre-configured
     * consent for the requested scopes. This can be used as a method to
     * check for existing authentication and/or consent.
     */
    const NONE = "none";
    /**
     * The Authorization Server MUST prompt the End-User for re-authentication
     */
    const LOGIN = "login";
    /**
     * The Authorization Server MUST prompt the End-User for consent before
     * returning information to the Client.
     */
    const CONSENT = "consent";
    /**
     * The Authorization Server MUST prompt the End-User to select a user account.
     * This allows a user who has multiple accounts at the Authorization Server to
     * select amongst the multiple accounts that they may have current sessions for.
     */
    const SELECT_ACCOUNT = "select_account";
    
    /**
     *  Parameter name
     * @var String 
     */
    private $paramName;
    
    /**
     * The constructor sets the parameter name
     * @param String Parameter
     */
    public function __construct($parameter = "") {
        $this->paramName = $parameter;
    }
    
    /**
     * Gets the parameter name entered
     * @return String Parameter name
     */
    public function getParamName() {
        return $this->paramName;
    }
    
    /**
     * Returns a string representation of the object. In this case the 
     * parameter name.
     * 
     * @return String Parameter name
     */
    public function toString(){
        return $this->paramName;
    }
    
    /**
     * Returns an array containing the constants of this enum type, in the 
     * order they are declared. This method may be used to iterate over the 
     * constants as follows:
     * 
     * <p>
     * foreach (Prompt::values() as $value) {
     * </p>
     * <p>
     *      return $value;
     * </p>
     * }
     * 
     * @return Array An array containing the constants of this enum type, in the 
     * order they are declared
     */
    public static function values(){
            return new \ArrayIterator(
                        array(
                            Prompt::CONSENT,
                            Prompt::LOGIN,
                            Prompt::NONE,
                            Prompt::SELECT_ACCOUNT
                        )
                    );
    }
    
}

?>

File: model/src/main/java/io/jans/as/model/common/Prompt.java
/*
 * Janssen Project software is available under the Apache License (2004). See http://www.apache.org/licenses/ for full text.
 *
 * Copyright (c) 2020, Janssen Project
 */

package io.jans.as.model.common;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

import java.util.ArrayList;
import java.util.List;

/**
 * An ASCII string values that specifies whether the Authorization Server
 * prompts the End-User for re-authentication and consent.
 *
 * @author Javier Rojas Blum Date: 02.10.2012
 */
public enum Prompt implements HasParamName {

    /**
     * The Authorization Server MUST NOT display any authentication or
     * consent user interface pages. An error is returned if the End-User
     * is not already authenticated or the Client does not have pre-configured
     * consent for the requested scopes. This can be used as a method to
     * check for existing authentication and/or consent.
     */
    NONE("none"),
    /**
     * The Authorization Server MUST prompt the End-User for re-authentication
     */
    LOGIN("login"),
    /**
     * The Authorization Server MUST prompt the End-User for consent before
     * returning information to the Client.
     */
    CONSENT("consent"),
    /**
     * The Authorization Server MUST prompt the End-User to select a user account.
     * This allows a user who has multiple accounts at the Authorization Server to
     * select amongst the multiple accounts that they may have current sessions for.
     */
    SELECT_ACCOUNT("select_account");

    private final String paramName;

    private Prompt(String paramName) {
        this.paramName = paramName;
    }

    /**
     * Returns the corresponding {@link Prompt} for a parameter
     * prompt of the authorization endpoint.
     *
     * @param param The parameter.
     * @return The corresponding response type if found, otherwise <code>null</code>.
     */
    public static Prompt fromString(String param) {
        if (param != null) {
            for (Prompt rt : Prompt.values()) {
                if (param.equals(rt.paramName)) {
                    return rt;
                }
            }
        }
        return null;
    }

    /**
     * Gets param name.
     *
     * @return param name
     */
    public String getParamName() {
        return paramName;
    }

    /**
     * Returns a list of the corresponding {@link Prompt} from a space-separated
     * list of prompt parameters.
     *
     * @param paramList A space-separated list of prompt parameters.
     * @param separator The separator of the string list.
     * @return A list of the recognized response types.
     */
    @JsonCreator
    public static List<Prompt> fromString(String paramList, String separator) {
        List<Prompt> prompts = new ArrayList<>();

        if (paramList != null && !paramList.isEmpty()) {
            String[] params = paramList.split(separator);
            for (String param : params) {
                for (Prompt p : Prompt.values()) {
                    if (param.equals(p.paramName) && !prompts.contains(p)) {
                        prompts.add(p);
                    }
                }
            }
        }

        return prompts;
    }

    /**
     * Returns a string representation of the object. In this case the parameter name.
     */
    @Override
    @JsonValue
    public String toString() {
        return paramName;
    }
}